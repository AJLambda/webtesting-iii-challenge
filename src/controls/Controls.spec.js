// Test away
import React from "react";
import renderer from "react-test-renderer"; // 1: install this npm module as a dev dependency
import { render } from "react-testing-library";

import Controls from "./Controls";

describe("<Controls />", () => {
  // 2. write this test
  it("runs the tests", () => {
    expect(true).toBe(true);
  });
  it("renders", () => {
    render(<Controls />);
  });
  it("matches snapshot", () => {
    const tree = renderer.create(<Controls />); // generates a DOM tree

    // snapshots are a JSON representation of the DOM tree
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
