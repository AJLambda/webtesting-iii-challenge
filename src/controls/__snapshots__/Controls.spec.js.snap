// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Controls /> The component matches snapshot 1`] = `
<div
  className="controls panel"
>
  <button
    className="toggle-btn"
    disabled={true}
  >
    Lock Gate
  </button>
  <button
    className="toggle-btn"
  >
    Close Gate
  </button>
</div>
`;

exports[`<Controls /> The entire Component matches snapshot 1`] = `
<div
  className="controls panel"
>
  <button
    className="toggle-btn"
    disabled={true}
  >
    Lock Gate
  </button>
  <button
    className="toggle-btn"
  >
    Close Gate
  </button>
</div>
`;

exports[`<Controls /> The entire component matches snapshot 1`] = `
<div
  className="controls panel"
>
  <button
    className="toggle-btn"
    disabled={true}
  >
    Lock Gate
  </button>
  <button
    className="toggle-btn"
  >
    Close Gate
  </button>
</div>
`;

exports[`<Controls /> matches snapshot 1`] = `
<div
  className="controls panel"
>
  <button
    className="toggle-btn"
    disabled={true}
  >
    Lock Gate
  </button>
  <button
    className="toggle-btn"
  >
    Close Gate
  </button>
</div>
`;
